src/Generating/FinishGen.cpp

```diff
@@ -818,22 +819,11 @@ void cFinishGenSprinkleFoliage::GenFinish(cChunkDesc & a_ChunkDesc)
 				{
 					float val3 = m_Noise.CubicNoise2D(xx * 0.01f + 10, zz * 0.01f + 10);
 					float val4 = m_Noise.CubicNoise2D(xx * 0.05f + 20, zz * 0.05f + 20);
-					if (val1 + val2 > 0.2f)
-					{
-						a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_YELLOW_FLOWER);
-					}
-					else if (val2 + val3 > 0.2f)
-					{
-						a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_RED_ROSE);
-					}
-					else if (val3 + val4 > 0.2f)
-					{
-						a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_RED_MUSHROOM);
-					}
-					else if (val1 + val4 > 0.2f)
-					{
-						a_ChunkDesc.SetBlockType(x, ++Top, z, E_BLOCK_BROWN_MUSHROOM);
-					}
+					if ((val1 + val2 > 0.2f) ||
+						(val2 + val3 > 0.2f) ||
+						(val3 + val4 > 0.2f) ||
+						(val1 + val4 > 0.2f))
+					{}
```

Comment:
This construction looks confusing, is it to remove the hardcoded block values?

Reply:
Yes, and to not alter the generation of the other blocks that are generated in the same function.

Reply:
I have tried it without this code. It seems to work as well. It is in the new commit.
Tall grass was also generated twice.